### PR TITLE
[ntuple] use RClusterPool in RNTupleMerger

### PR DESCRIPTION
**Pull request by silverweed**
# This Pull request:
adds the use of `RClusterPool` to preload clusters during merging of RNTuples.

## Checklist:

- [x] tested changes locally
- [ ] updated the docs (if necessary)

### CODE REVIEW

File: tree/ntuple/v7/src/RNTupleMerger.cxx [hunk 1]
```diff
@@ -203,9 +215,10 @@ void ROOT::Experimental::Internal::RNTupleMerger::Merge(std::span<RPageSource *>
       auto clusterId = descriptor->FindClusterId(0, 0);
 
       while (clusterId != ROOT::Experimental::kInvalidDescriptorId) {
-         auto &cluster = descriptor->GetClusterDescriptor(clusterId);
+         auto *clusterP = clusterPool.GetCluster(clusterId, columnSet);
+         assert(clusterP);
+         const auto &cluster = descriptor->GetClusterDescriptor(clusterId);
```
Comment (jblomer) [inline tree/ntuple/v7/src/RNTupleMerger.cxx:220]: Perhaps for clarity:
```suggestion
         auto *clusterPool = clusterPool.GetCluster(clusterId, columnSet);
         assert(clusterPool);
         const auto &clusterDesc = descriptor->GetClusterDescriptor(clusterId);
```

**Review by jblomer — CHANGES_REQUESTED** [review]
In `AddColumnsFromField` and `CollectColumns` we now produce both, the vector of `RColumnInfo` records and the `RCluster::ColumnSet_t`. To me it seems that the source of truth is the `RColumnInfo` vector, and it may be clearer to generate and `RCluster::ColumnSet_t` from the `RColumnInfo` vector.

File: tree/ntuple/v7/src/RNTupleMerger.cxx (hunk 1 above)
Comment (silverweed) [reply tree/ntuple/v7/src/RNTupleMerger.cxx:220]: I guess you meant `cluster` rather than `clusterPool`, since `clusterPool` already exists

**Comment by silverweed** [comment]
> In `AddColumnsFromField` and `CollectColumns` we now produce both, the vector of `RColumnInfo` records and the `RCluster::ColumnSet_t`. To me it seems that the source of truth is the `RColumnInfo` vector, and it may be clearer to generate and `RCluster::ColumnSet_t` from the `RColumnInfo` vector.

It was a small optimization to avoid looping over all columns twice and get all the data we need at once. I can separate the loops if you prefer, it should make little difference.

File: tree/ntuple/v7/src/RNTupleMerger.cxx (hunk 1 above)
Comment (jblomer) [reply tree/ntuple/v7/src/RNTupleMerger.cxx]: Oh. Yes :)

**Review by jblomer — APPROVED** [review]

**Pull request merged**